### PR TITLE
Tall fences and walls

### PR DESCRIPTION
**Pull request by mbartlett21**
This PR adds an option in settings to make fences and walls invisibly taller by 3/8 of a node, not allowing the player to jump over them.

I have also edited the collision box to allow players to run along fences and walls, rather than hitting into the poles

### CODE REVIEW

**Comment by sfan5** [comment]
I've often been wondering why we don't have this.
If it was up to be me, this wouldn't even need to be an option. But people may want to disable it, so that's fine.

**Comment by mbartlett21** [comment]
Just by the way, if you place snow on top of the fence, you can then jump over it where the snow is.
Fences stacked on top of each other also don't have a problem

**Comment by paramat** [comment]
Logically, our walls and fences are 1 node high, players can jump over 1 node high obstacles so it is natural to be able to jump over walls and fences.
However, this could be useful, and has been requested, for keeping animals trapped inside.
It needs to be a optional, and default disabled, to not break existing worlds.

File: mods/default/functions.lua [hunk 1]
```diff
@@ -303,6 +304,16 @@ function default.register_fence(name, def)
...
 		},
+		collision_box = {
+			type = "connected",
+			fixed = {-1/8, -1/2, -1/8, 1/8, 1/2 + fence_collision_extra, 1/8},
+			-- connect_top =
+			-- connect_bottom =
+			connect_front = {-1/8,-1/2,-1/2,1/8,1/2 + fence_collision_extra,-1/8},
+			connect_left = {-1/2,-1/2,-1/8,-1/8,1/2 + fence_collision_extra,1/8},
+			connect_back = {-1/8,-1/2,1/8,1/8,1/2 + fence_collision_extra,1/2},
+			connect_right = {1/8,-1/2,-1/8,1/2,1/2 + fence_collision_extra,1/8},
```
Comment (MoNTE48) [inline mods/default/functions.lua]: - connect_right = {1/8,-1/2,-1/8,1/2,1/2 + fence_collision_extra,1/8},
+ connect_right = {1/8,-1/2,-1/8,1/2,1/2 + fence_collision_extra, 1/8}
(space and comma)

**Comment by TumeniNodes** [comment]
Don't neglect fence rails.

**Comment by mbartlett21** [comment]
> Don't neglect fence rails.
> This will make no sense if they are not included with the option

@TumeniNodes 
Fence rails are included in this PR :)

File: mods/default/functions.lua (hunk 1 above)
Comment (mbartlett21) [reply mods/default/functions.lua]: What is wrong with this?
It follows the style that this file was using in keeping the last comma, even though it is not needed.

Comment (MoNTE48) [reply mods/default/functions.lua]: Well, it wasn't bad to fix the style. Although this will not change anything. :)

**Comment by DS-Minetest** [comment]
>I have also edited the collision box to allow players to run along fences and walls, rather than hitting into the poles

This might feel weird.
Can't the old behaviour be completely preserved?

**Comment by sfan5** [comment]
Tested and works :+1:

> I have also edited the collision box to allow players to run along fences and walls
Can you also do this for fence gates?

**Comment by mbartlett21** [comment]
@sfan5 Done!

**Comment by mbartlett21** [comment]
> It needs to be a **optional, and default disabled**, to not break existing worlds.

@paramat 
Currently, I have added a setting (called Tall fences and walls) that is disabled by default. Is its location okay?

File: mods/default/functions.lua (hunk 1 above)
Comment (mbartlett21) [reply mods/default/functions.lua]: @MoNTE48
Fixed!

**Comment by paramat** [comment]
Yes i saw the setting and the default, thanks.
The location is fine, that file doesn't seem to have any logical order anyway.

**Comment by paramat** [comment]
You also need to add the setting to MTG's 'minetest.conf_example' file.

**Comment by mbartlett21** [comment]
@paramat Done!

**Comment by paramat** [comment]
@mbartlett21 players can still jump over using sneak-jump, so this ends up being for animals only, is that part of the intention?

**Comment by mbartlett21** [comment]
Yes

**Comment by paramat** [comment]
> I have also edited the collision box to allow players to run along fences and walls, rather than hitting into the poles

Good idea.

**Comment by mbartlett21** [comment]
:hooray:

**Pull request merged**